diff -urN mbed.orig/workspace_tools/tests.py mbed/workspace_tools/tests.py
--- mbed.orig/workspace_tools/tests.py	2015-03-18 10:02:30.241958595 +0100
+++ mbed/workspace_tools/tests.py	2015-03-18 10:17:39.805942307 +0100
@@ -819,6 +819,16 @@
         "host_test": "udp_link_layer_auto",
         "peripherals": ["ethernet"],
     },
+    {
+        "id": "NET_14", "description": "TFTPServer",
+        "source_dir": join(TEST_DIR, "net", "protocols", "TFTPServer"),
+        "dependencies": [MBED_LIBRARIES, RTOS_LIBRARIES, ETH_LIBRARY,  SD_FS, FAT_FS],
+    },
+    {
+        "id": "laser", "description": "LaosLaser",
+        "source_dir": join(TEST_DIR, "net", "protocols", "laser"),
+        "dependencies": [MBED_LIBRARIES, RTOS_LIBRARIES, ETH_LIBRARY], #,  SD_FS, FAT_FS],
+    },
 
     # u-blox tests
     {
